### PR TITLE
Add Navbar and skeleton pages

### LINKED ISSUE #12: FE: Dashboard Skeleton
- Adapt the current project skeleton to match project mockups
- The project skeleton will have the general frontend structure of the project, but without the specific components filled out
- If there is a navbar in the mockups, the skeleton should include the navbar + content pages for the app for a logged in user experience
- The navbar should be persistent across the whole logged in user experience and include a placeholder image for the user
- Dropdown for profile in navbar should show two links:
    - logout (already implemented on the backend)
    - go to profile

### PR DESCRIPTION
**Pull request by joelmackenz**
### What this PR does (required):
- Adds a navbar component and skeleton pages for MyJobs, Notifications, Messages, and Profile. The Navbar includes links to all of these, and dynamically renders information for the top links, which displays upon hover. Dummy info is provided at the top in order to populate the fields. The links will navigate to their respective pages. The Profile photo is clickable, and opens a popover for the profile page and logout page (logout was not implemented, though: it simply leads to an empty function). 

### Screenshots / Videos (required):
- 
Screen Shot 2021-06-01 at 10 31 48 PM
Screen Shot 2021-06-01 at 10 31 55 PM

### Any information needed to test this feature (required):
- A Navbar component must be imported into App.js, as well as paths to the new pages. I plan on including these once I have access to the protected routes, to keep the navbar separate from the unprotected routes.

### Any issues with the current functionality (optional):
-I provided a null key to all items being mapped on line 131 of Navbar.tsx. I tried to make the key equal the name of the mapped item ("alert", in this case), but this causes TypeScript to throw "No overload matches this call." I'm not sure how to give the items unique keys in this circumstance.

### CODE REVIEW

**Comment by g4rry420** [comment]
#### The styling looks good Joel.
#### Just a couple of suggestions according to me:
- Currently, when I hover the `Notifications` tab or if I clicked the `Profile` avatar there is `padding-right` coming up to the header of navbar. Unfortunately, this is currently [bug](https://github.com/mui-org/material-ui/issues/18664) in material-ui. I believe this can solved using `Popper` instead of `Popover`.
- `useStyles` hook should be in separate file to make it consistent with rest of the project structure.
- `Notifications` tab shouldn't be wrapped inside Link tag as user is not going to send over to different page. You can `Typography` in case.
- The navbar isn't responsive. I recommend to check out [this](https://betterprogramming.pub/making-a-basic-header-responsive-with-materialui-and-react-2198fac923c8) article. But, instead of using custom useState for mobile screen detection you can use [useMediaquery](https://material-ui.com/components/use-media-query/#usemediaquery) provided by material-ui out of the box.

Let me know if you need any help :)

File: client/src/components/Layout/Navbar.tsx [hunk 1]
```diff
@@ -0,0 +1,217 @@
...
+        vertical: 'top',
+        horizontal: 'center',
+      }}
+      className={classes.popover}
+    >
+      <MenuList autoFocusItem={Boolean(titleAnchor)}>
+        <MenuItem>
+          <Link className={classes.linkItem} to="/notifications">
+            {activeNotifications.map((notification) =>
+              notification.title !== null ? (
```
Comment (g4rry420) [inline client/src/components/Layout/Navbar.tsx]: Here you can get `(notifications, index)` from the map and give the `key={index}` for now. Later, when fetching from backend, I believe we can give `id` to key.

**Review by g4rry420 — COMMENTED** [review]
#### The styling looks good Joel.
#### Just a couple of suggestions according to me:
- Currently, when I hover the `Notifications` tab or if I clicked the `Profile` avatar there is `padding-right` coming up to the header of navbar. Unfortunately, this is currently [bug](https://github.com/mui-org/material-ui/issues/18664) in material-ui. I believe this can solved using `Popper` instead of `Popover`.
- `useStyles` hook should be in separate file to make it consistent with rest of the project structure.
- `Notifications` tab shouldn't be wrapped inside Link tag as user is not going to send over to different page.
- The navbar isn't responsive. I recommend to check out [this](https://betterprogramming.pub/making-a-basic-header-responsive-with-materialui-and-react-2198fac923c8) article. But, instead of using custom useState for mobile screen detection you can use [useMediaquery](https://material-ui.com/components/use-media-query/#usemediaquery) provided by material-ui out of the box.

Let me know if you need any help :)

File: client/src/components/Layout/Navbar.tsx [hunk 2]
```diff
@@ -0,0 +1,217 @@
...
+          </Link>
+        </MenuItem>
+      </MenuList>
+    </Popover>
+  );
+
+  return (
+    <AppBar style={{ position: 'static' }}>
+      <Toolbar className={classes.navMain}>
+        <img src={logo} className={classes.logo} />
```
Comment (g4rry420) [inline client/src/components/Layout/Navbar.tsx]:
```suggestion
        <img src={logo} alt='Logo' className={classes.logo} />
```
For web accessibility

File: client/src/components/Layout/Navbar.tsx [hunk 3]
```diff
@@ -0,0 +1,217 @@
...
+          <Link className={classes.linkItem} onClick={handleLogout} to="/">
+            Logout
+          </Link>
+        </MenuItem>
+      </MenuList>
+    </Popover>
+  );
+
+  return (
+    <AppBar style={{ position: 'static' }}>
```
Comment (bonnieli) [inline client/src/components/Layout/Navbar.tsx]: Prefer to put this in useStyles just to stay consistent with styling method

File: client/src/components/Layout/Navbar.tsx [hunk 4]
```diff
@@ -0,0 +1,217 @@
...
+                to={`/notifications`}
+                className={classes.linkItem}
+                aria-haspopup="true"
+                onMouseOver={(e) => setNotificationsAnchor(e.currentTarget)}
+                onMouseLeave={() => setNotificationsAnchor(null)}
+              >
+                Notifications
+              </Link>
+            </MenuItem>
+            <MenuItem key={'myjobs'}>
```
Comment (bonnieli) [inline client/src/components/Layout/Navbar.tsx]: is a key needed here?

**Review by bonnieli — COMMENTED** [review]
nice clean code!

**Review by bonnieli — APPROVED** [review]

File: client/src/components/Layout/Navbar.tsx (hunk 4 above)
Comment (joelmackenz) [reply client/src/components/Layout/Navbar.tsx]: I can't remember why I did that. Haha

**Comment by joelmackenz** [comment]
@g4rry420 Thanks for the thorough review, Gurkiran.
I also noticed the padding-right issue with the popover, although it only seemed to happen on the pre-loaded dashboard page. I refactored them all to poppers, and these do work much better. Padding-right issue is removed.
I made the navbar responsive. It's pretty simple right now, but it works, and it'll let us move forward. It can be expanded in the future if we have enough time.
Finally, I was picturing there being a notifications page, similar to the one on Facebook. We should all chat about this during the next meeting. I kept the Notifications page and link in, for the time being.

**Comment by joelmackenz** [comment]
As I was refactoring anyway, I updated the profile menu to work as a toggle. @bonnieli, does the approval still stand? It's a small change, but I just want to be sure :)

**Comment by bonnieli** [comment]
> As I was refactoring anyway, I updated the profile menu to work as a toggle. @bonnieli, does the approval still stand? It's a small change, but I just want to be sure :)

yep approval still stands!

**Comment by joelmackenz** [comment]
Thank you!

**Pull request merged**